tensorboard/uploader/exporter.py

```diff
@@ -115,10 +118,40 @@ def export(self, read_time=None):
         """
         if read_time is None:
             read_time = time.time()
-        for experiment_id in self._request_experiment_ids(read_time):
+        experiment_metadata_mask = experiment_pb2.ExperimentMask(
+            create_time=True, update_time=True, name=True, description=True,
+        )
+        experiments = list_experiments(
+            self._api, fieldmask=experiment_metadata_mask, read_time=read_time
+        )
+        for experiment in experiments:
+            if isinstance(experiment, experiment_pb2.Experiment):
+                experiment_id = experiment.experiment_id
+                experiment_metadata = {
+                    "name": experiment.name,
+                    "description": experiment.description,
+                    "create_time": util.format_time_absolute(
+                        experiment.create_time
+                    ),
+                    "update_time": util.format_time_absolute(
+                        experiment.update_time
+                    ),
+                }
+            elif isinstance(experiment, str):
```

Comment:
Are there actually anymore of these old servers that stream experimen_id instead of experiment proto?

Reply:
Good question—no, there shouldn’t be. I was considering what to do about
this and thought that it was easier and a bit more conservative to keep
the code as is (this is moved out of `_request_experiment_ids`) than to
change `list_experiments` below and drop the old code path. But you’re
right that this has been deployed for a while, so it looks unlikely that
we’ll need to roll the server back. I can go ahead and make that change.

Reply:
Done as a pre-change in #3310, which indeed simplifies this diff by a
fair amount.